engineering_tools/carma_script_extensions/configure-cloud-sim.bash

```diff
@@ -0,0 +1,205 @@
+#  use this file except in compliance with the License. You may obtain a copy of
+#  the License at
+# 
+#  http://www.apache.org/licenses/LICENSE-2.0
+# 
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#  License for the specific language governing permissions and limitations under
+#  the License.
+
+# Function to display help message
+display_help() {
+        cat <<HELP
+-------------------------------------------------------------------------------
+| USDOT FHWA STOL CARMA                                              |
+-------------------------------------------------------------------------------
+Usage: $0 {init|stop|help}
+Please enter one of the following commands when using the configure-cloud-sim extension:
+        init     - Create a VPC with internet gateway and route table"
+        stop     - Delete the created VPC, internet gateway, and route table"
+        help     - Display this help message"
+HELP
+}
+
+# Function to create the VPC, internet gateway, and route table
+create_resources() {
+
+    if [ ! command -v "aws" &> /dev/null ]; then
+        echo "aws cli is not installed. Attempting to install. This requires sudo."
+        
+        curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+        unzip awscliv2.zip
+        sudo ./aws/install
+
+        echo "NOTE: You need to manually configure the aws cli for your user access key. Refer to this video for instructions https://www.youtube.com/watch?v=Rp-A84oh4G8"
+        exit 1
+    fi
+    
+    # Try deleting XiL Cloud resources if any exist
+    echo "Checking and deleting existing XiL-Cloud resources first"
+    delete_resources
+
+    vpc_success=0
+    igw_success=0
+    route_table_success=0
+
+    # Set your desired CIDR block here
+    CIDR_Block="10.0.0.0/16"
+
+    # Create VPC
+    vpc_id=$(aws ec2 create-vpc --cidr-block $CIDR_Block --tag-specifications 'ResourceType=vpc,Tags=[{Key=Name,Value=XiL-Cloud VPC}]' \
+    --query 'Vpc.VpcId' --output text) 
+
+    # Check the exit code to determine if the VPC creation was successful
+    if [ $? -eq 0 ]; then
+        vpc_success=1
+        echo "Created vpc: $vpc_id"
+    else
+        echo "Error creating XiL-Cloud VPC: $vpc_id"
+    fi
+
+    # Create Internet Gateway
+    igw_id=$(aws ec2 create-internet-gateway \
+    --tag-specifications 'ResourceType=internet-gateway,Tags=[{Key=Name,Value=XiL-Cloud IGW}]' --query 'InternetGateway.InternetGatewayId' \
+    --output text) 
+
+    # Check the exit code to determine if the Internet Gateway creation was successful
+    if [ $? -eq 0 ]; then
+        igw_success=1
+        echo "Created Internet Gateway: $igw_id"
+    else
+        echo "Error creating Internet Gateway: $igw_id"
```

Comment:
I think in each failure case here, we should clean up all the resources we'd allocated before the failure, then exit the script so we don't allocate any more. Ideally the only possible outcomes of this script are "successful fully initialized AWS resources" or "nothing happened".

Reply:
Yup thats a good point. In a case where we failed to set up one of the resources, configuration has faile